syft/rekor/rekor.go

```diff
@@ -0,0 +1,122 @@
+package rekor
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/source"
+	"github.com/sigstore/cosign/cmd/cosign/cli/rekor"
+	"github.com/sigstore/rekor/pkg/generated/client"
+	"github.com/spdx/tools-golang/spdx"
+)
+
+const (
+	DefaultRekorAddr = "https://rekor.sigstore.dev"
+	InfoForUser      = `
+			[EXPERIMENTAL FEATURE: Rekor-cataloger] 
+			
+			This SBOM contains a relationship that references an external document. This 
+			document is not present in the cataloged image or directory; rather it has 
+			been found by searching the Rekor transparency log (https://www.sigstore.dev/).  
+			
+			Trusting this external document relationship requires trusting several entities: 
+				- the user or CI/CD action that uploaded an entry to Rekor
+				- Rekor transparency log
+				- Fulcio CA
+
+			The Rekor entry(s) that were used to create the external document relationship(s)
+			are listed below by UUID. See https://github.com/sigstore/rekor for 
+			information on how to query Rekor. 
+			`
+)
+
+type Client struct {
+	rekorClient *client.Rekor
+	httpClient  *http.Client
+}
+
+type ExternalRef struct {
+	SpdxRef spdx.ExternalDocumentRef2_2
+}
+
+// NewClient returns a client to use the rekor library
+func NewClient() (*Client, error) {
+	rekorClient, err := rekor.NewClient(DefaultRekorAddr)
+	if err != nil {
+		return nil, errors.New("error creating rekor client")
+	}
+
+	return &Client{
+		rekorClient: rekorClient,
+		httpClient:  &http.Client{},
+	}, nil
+}
+
+func (r ExternalRef) ID() artifact.ID {
+	id, err := artifact.IDByHash(r.SpdxRef.Checksum)
+	if err != nil {
+		panic("id could not be created from hash for an external ref")
```

Comment:
yea.. think panic is probably bad here. It would make sense that ID() would not be able to throw an error by construction... however, the calling functions do return errors... Maybe follow this for now:

https://github.com/anchore/syft/blob/f24bbc1838351e4e408368f3312c76db6552e71e/syft/source/coordinates.go#L19-L20